enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -11780,8 +11782,7 @@ class AdjointGenerator
       }
 
       assert(newcalled);
-      FunctionType *FT =
-          cast<FunctionType>(newcalled->getType()->getPointerElementType());
+      assert(FT);
```

Comment:
this assertion will never fail.

Reply:
True, but I'd still like to have it as an assertion.